kotsadm/web/src/components/apps/AppVersionHistoryRow.jsx

```diff
@@ -42,29 +42,35 @@ function renderVersionAction(version, nothingToCommitDiff, app, history, deployV
   }
 
   const isCurrentVersion = version.sequence === downstream.currentVersion?.sequence;
+  const isDeploying = version.status === "deploying";
   const isPastVersion = find(downstream.pastVersions, { sequence: version.sequence });
   const needsConfiguration = version.status === "pending_config";
   const showActions = !isPastVersion || app.allowRollback;
   const isSecondaryBtn = isPastVersion || needsConfiguration;
   const isRollback = isPastVersion && version.deployedAt && app.allowRollback;
+  const isRedeploy = isCurrentVersion && version.status === "failed";
 
   return (
     <div className="flex flex1 justifyContent--flexEnd">
       {showActions &&
         <button
-          className={classNames("btn", { "secondary blue": isSecondaryBtn, "primary blue": !isSecondaryBtn })}
-          disabled={isCurrentVersion}
+          className={classNames("btn", { "secondary dark": isRollback, "secondary blue": isSecondaryBtn || isRedeploy && !isRollback, "primary blue": !isSecondaryBtn && !isRedeploy && !isRollback })}
+          disabled={isCurrentVersion && version.status !== "failed"}
           onClick={() => needsConfiguration ? history.push(`/app/${app.slug}/config/${version.sequence}`) : deployVersion(version)}
         >
           {needsConfiguration ?
             "Configure" :
             downstream.currentVersion?.sequence == undefined ?
               "Deploy" :
-              isRollback ?
-                "Rollback" :
-                isCurrentVersion ?
-                  "Deployed" :
-                  "Deploy"
+              isRedeploy ?
```

Comment:
Can we set this label a different way? This is really hard to grok when reading.

Reply:
Open for ideas :) 
I thought moving this logic in separate function and calling it here but that's not changing the label. It can just clean up these ternary operators and make code cleaner.